src/compiler/scala/tools/nsc/classpath/FileUtils.scala

```diff
@@ -32,8 +32,11 @@ object FileUtils {
   implicit class FileOps(val file: JFile) extends AnyVal {
     def isPackage: Boolean = file.isDirectory && mayBeValidPackage(file.getName)
 
-    def isClass: Boolean = file.isFile && file.getName.endsWith(".class")
+    def isClass: Boolean = file.isFile && endsClass(file.getName)
   }
+  private val SUFFIX_CLASS = ".class"
+  private val SUFFIX_SCALA = ".scala"
+  private val SUFFIX_JAVA = ".java"
```

Comment:
Java coders shout their constant names out of frustration, the constant frustration.